sunpy/net/hek/hek.py

```diff
@@ -17,30 +15,89 @@
 from sunpy.net import attr
 from sunpy.net.base_client import BaseClient, QueryResponseTable
 from sunpy.net.hek import attrs
-from sunpy.time import parse_time
+from sunpy.net.hek.utils import freeze, parse_times, parse_values_to_quantities
 from sunpy.util import dict_keys_same, unique
 from sunpy.util.xml import xml_to_dict
 
 __all__ = ['HEKClient', 'HEKTable', 'HEKRow']
 
 DEFAULT_URL = 'https://www.lmsal.com/hek/her?'
 
-
-def _freeze(obj):
-    """ Create hashable representation of result dict. """
-    if isinstance(obj, dict):
-        return tuple((k, _freeze(v)) for k, v in obj.items())
-    if isinstance(obj, list):
-        return tuple(_freeze(elem) for elem in obj)
-    return obj
-
-
 class HEKClient(BaseClient):
     """
     Provides access to the Heliophysics Event Knowledgebase (HEK).
 
     The HEK stores solar feature and event data generated by algorithms and
     human observers.
+
+    Examples
+    --------
+    >>> from sunpy.net import hek
+    >>> from sunpy.net import attrs as a
+    >>> tstart = '2011/08/09 07:23:56'
+    >>> tend = '2011/08/09 12:40:29'
+    >>> event_type = 'FL'
+    >>> client = hek.HEKClient()
+    >>> result = client.search(a.Time(tstart,tend), a.hek.EventType(event_type), (a.hek.Event.Coord1 > 50) and (a.hek.FL.PeakFlux > 1000))  # doctest: +REMOTE_DATA
+    >>> result['fl_peakflux']  # doctest: +REMOTE_DATA
+    <QueryResponseColumn name='fl_peakflux' dtype='object' length=7>
+    2326.86 DN / (pix s)
+    1698.83 DN / (pix s)
+    2360.49 DN / (pix s)
+    3242.64 DN / (pix s)
+    1375.93 DN / (pix s)
+    6275.98 DN / (pix s)
+    1019.83 DN / (pix s)
+
```

Comment:
```suggestion
```

Honestly I am not a major fan of examples in client doc strings.

I think the rendered RST pages are better for this.